include/pthreadpool.h

```diff
@@ -85,6 +85,17 @@ pthreadpool_t pthreadpool_create(size_t threads_count);
  */
 size_t pthreadpool_get_threads_count(pthreadpool_t threadpool);
 
+/*
+ * API to enable doing work with fewer threads than available in
+ * threadpool.
+ * This API takes in a pointer to threadpool object and sets max
+ * number of threads to use with that threadpool.
+ * Purpose of this is to ameliorate some perf degradation observed
+ * due to OS mapping a given set of threads to fewer cores.
+ */
+void pthreadpool_set_max_num_threads(struct pthreadpool* threadpool, size_t num_threads);
+size_t pthreadpool_get_max_num_threads();
```

Comment:
This function should have `void` in the parameter list for compatibility with C

Reply:
Aah good to know. Did not know this.